### PR TITLE
migrate to buf

### PR DESCRIPTION
**Pull request by jzelinskie**

### CODE REVIEW

File: buf.gen.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env -S buf generate buf.build/authzed/api:3444bd6f843d49928d6a1226965d88b7 --template
+version: "v1beta1"
```
Comment (samkim) [inline buf.gen.yaml:2]: Is this supposed to be `v1alpha1`?

File: lib/authzed/api/v0/acl_service_pb.rb [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    end
+    add_message "authzed.api.v0.LookupResponse" do
+      repeated :resolved_object_ids, :string, 1, json_name: "resolvedObjectIds"
+      optional :next_page_reference, :string, 2, json_name: "nextPageReference"
+      optional :revision, :message, 3, "authzed.api.v0.Zookie", json_name: "revision"
+    end
+  end
+end
+
+module Authzed
```
Comment (samkim) [inline lib/authzed/api/v0/acl_service_pb.rb:82]: This resulting module structure is much better than previous

File: buf.gen.yaml (hunk 1 above)
Comment (jzelinskie) [reply buf.gen.yaml:2]: Nope -- this is the version of the schema for this file. Joey actually thought the same thing earlier today.

**Review by samkim — APPROVED** [review]
LGTM

**Pull request merged**